### PR TITLE
Re-order list of resolvers according to their searchOrder

### PR DESCRIPTION
**Pull request by gwre-ivan**
Motivation:
Mac OS specific DNS resolver fails to take into account search order
of resolvers causing wrong resolver being used is some circumstances

Modifications:
Re-order array of resolvers using their sort order as an ordering key.
Final order is opposite of the search order to make sure that resolver
with the lower sort order goes last (so it overrides previous one
in the `resolverMap`).

Result:
Fixes issue https://github.com/netty/netty/issues/11225

Motivation:

Explain here the context, and why you're making that change.
What is the problem you're trying to solve.

Modification:

Describe the modifications you've done.

Result:

Fixes #<GitHub issue number>. 

If there is no issue then describe the changes introduced by this PR.

### CODE REVIEW

**Comment by normanmaurer** [comment]
@gwre-ivan did you sign our icla:

https://netty.io/s/icla

File: resolver-dns-native-macos/src/main/java/io/netty/resolver/dns/macos/MacOSDnsServerAddressStreamProvider.java [hunk 1]
```diff
@@ -120,6 +130,7 @@ public MacOSDnsServerAddressStreamProvider() {
         if (resolvers == null || resolvers.length == 0) {
             return Collections.emptyMap();
         }
+        Arrays.sort(resolvers, Collections.reverseOrder(RESOLVER_COMPARATOR));
```
Comment (normanmaurer) [inline resolver-dns-native-macos/src/main/java/io/netty/resolver/dns/macos/MacOSDnsServerAddressStreamProvider.java:133]: Store the instance that is returned by reverseOrder in the static field directly and reuse ?

**Review by normanmaurer — CHANGES_REQUESTED** [review]

File: resolver-dns-native-macos/src/main/java/io/netty/resolver/dns/macos/MacOSDnsServerAddressStreamProvider.java (hunk 1 above)
Comment (normanmaurer) [reply resolver-dns-native-macos/src/main/java/io/netty/resolver/dns/macos/MacOSDnsServerAddressStreamProvider.java:133]: yep ... whatever fits you best.

Comment (gwre-ivan) [reply resolver-dns-native-macos/src/main/java/io/netty/resolver/dns/macos/MacOSDnsServerAddressStreamProvider.java]: Done.

**Comment by gwre-ivan** [comment]
Signed.

**Comment by normanmaurer** [comment]
@gwre-ivan please fix the check style error:

```
Error:  /home/runner/work/netty/netty/resolver-dns-native-macos/src/main/java/io/netty/resolver/dns/macos/MacOSDnsServerAddressStreamProvider.java:52: Line is longer than 120 characters (found 128). [LineLength]
5083
Audit done.
```

**Comment by normanmaurer** [comment]
@gwre-ivan thanks a lot!

**Pull request merged**